components/eventing-controller/pkg/backend/jetstreamv2/jetstream.go

```diff
@@ -39,32 +41,178 @@ const (
 	originalTypeHeaderName = "originaltype"
 )
 
-func NewJetStream(config backendnats.Config, metricsCollector *backendmetrics.Collector,
-	cleaner cleaner.Cleaner, subsConfig env.DefaultSubscriptionConfig, logger *logger.Logger) *JetStream {
+func NewJetStream(config backendnats.Config, metricsCollector *backendmetrics.Collector, cleaner cleaner.Cleaner,
+	subsConfig env.DefaultSubscriptionConfig, logger logger.KLogger) *JetStream {
 	return &JetStream{
-		Config:           config,
-		logger:           logger,
-		subscriptions:    make(map[SubscriptionSubjectIdentifier]Subscriber),
-		metricsCollector: metricsCollector,
-		cleaner:          cleaner,
-		subsConfig:       subsConfig,
+		logger:            logger,
+		subscriptions:     make(map[SubscriptionSubjectIdentifier]Subscriber),
+		metricsCollector:  metricsCollector,
+		cleaner:           cleaner,
+		connectionBuilder: NewConnectionBuilder(config),
+		config:            config,
+		ceClientFactory:   cloudevent.ClientFactory{},
+		subsConfig:        subsConfig,
 	}
 }
 
-func (js *JetStream) Initialize(connCloseHandler backendutilsv2.ConnClosedHandler) error {
-	if err := js.validateConfig(); err != nil {
+// TODO(nils): In theory I don't see a reason why the integration tests need to know about the connection status of
+// the backend. Talk to @raypinto how we can resolve this problem.
+// To keep the tests green, for now I leave it as it is.
+func (js *JetStream) IsConnected() bool {
+	return js.connection.IsConnected()
+}
+
+// TODO(nils): I don't see a reason why the integration tests need to get the config from the backend. Since the tests
+// start the backend, they should already be aware of the config.
+// To keep the tests green, for now I leave it as it is.
+// Another solution might be to use a singleton for the config and don't access it from the JetStream backend.
+func (js *JetStream) GetConfig() *backendnats.Config {
```

Comment:
why do we return a reference here? in all testcases where this is used, the reference is immediately dereferenced. just return the object by value and the code would be easier to read